### PR TITLE
[bitmanip] Generate cmov under ZBT

### PR DESCRIPTION
**Pull request by rdolbeau**
Add basic support for cmov (ZBT), for [not]equal 0.

Signed-off-by: Romain Dolbeau <romain.dolbeau@european-processor-initiative.eu>

### CODE REVIEW

**Comment by kito-cheng** [comment]
Thanks, it seems like missing testcase, but I can add it later.

**Comment by kito-cheng** [comment]
Let me run full gcc testsuite before merge.

**Comment by rdolbeau** [comment]
@kito-cheng I added a test case for cmov.

File: gcc/testsuite/gcc.target/riscv/rvb-zbt-cmov.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        return (s == 0 ? b : c);
+}
+#else
+long cmov(long s, long b, long c) {
+        return (s ? b : c);
+}
+long cmov2(long s, long b, long c) {
+        return (s == 0 ? b : c);
+}
+#endif
```
Comment (kito-cheng) [inline gcc/testsuite/gcc.target/riscv/rvb-zbt-cmov.c:19]: All using long is fine :)

**Comment by kito-cheng** [comment]
@rdolbeau let you know, I'll verify this PR and #265 today, and then #267 later, thanks you :)

**Comment by rdolbeau** [comment]
@kito-cheng Of course you're right, `long` is fine. Fixed this and #265

**Comment by rdolbeau** [comment]
> @rdolbeau let you know, I'll verify this PR and #265 today, and then #267 later, thanks you :)

#267 isn't ready yet unfortunately; there's [at least] one bug I'm not able to figure out :-(

**Pull request merged**